sktime/datatypes/_vectorize.py

```diff
@@ -285,9 +285,9 @@ def _get_X_at_index(self, row_ind=None, col_ind=None, X=None):
         elif row_ind is None:
             res = X[col_ind]
         else:
-            res = X[col_ind].loc[row_ind]
+            res = X.loc[row_ind, col_ind]
         res = _enforce_index_freq(res)
-        return res
+        return res.copy()
```

Comment:
why do we add a `copy` here? genuinely asking.

Reply:
I mentioned it in the description. Without this, there is a memory leak which I can't really explain. I saw memory increasing above 50G (for ~100M parquet input data) until the python process got killed by the OS. It's probably pandas copying the frame for some reason. Pandas is really unclear to me, when it comes to returning copies vs refs. This observation is actually what started my journey.

Reply:
This is a bit dangerous as it is in a central location.
It could be something specific about your system.

Could you provide an example so I can verify the memory leak?
We should also raise a bug report then (I can also write it if you give a code example)

Reply:
Here you go. The generated data is similar to what mine looks like.

```
import os

import numpy as np
import pandas as pd
import psutil
from sktime.datatypes import VectorizedDF

index = pd.MultiIndex.from_product([
    np.arange(1000), pd.period_range(end="2022-12-31", periods=365 * 5, freq="D")
], names=["store", "date"])

data = np.random.default_rng().standard_normal((len(index), 50), dtype=np.float32)

df = pd.DataFrame(data=data, index=index)

process = psutil.Process(os.getpid())
print("{mem:.3f} GiB".format(mem=process.memory_info().rss / 1024 ** 3))  # ~0.4GiB

VectorizedDF(df).as_list()
```

This script runs until killed and returns with exit code 137 (oom).
`python oom.py; print $?`

Reply:
ouch. Thanks, although it seems mysterious to me.